servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/DefaultBlockingIterableProcessor.java

```diff
@@ -144,43 +119,37 @@ public boolean hasNext() {
                 return true;
             }
 
-            final Object next;
+            final boolean consumed;
             try {
-                next = buffer.take();
+                consumed = buffer.consume(this);
             } catch (InterruptedException e) {
                 return throwException(e);
             }
-            return processHasNext(next);
+            return terminal == null ? consumed : hasNextWhenTerminated();
```

Comment:
nit: this is the same as line 83. Maybe share ?

Reply:
mmm .. its a one-liner which uses a method anyways. I will avoid creating a method for it